### ISSUE TITLE
GPU memory leak

### ISSUE THREAD

**Issue by pzdkn** [KempnerInstitute/overcomplete#5]
In `log_metrics.py`, there is a GPU memory leak when we use `monitoring=2`:

```
def _log_metrics(monitoring, logs, model, z, loss, optimizer):
   ...
    if monitoring > 1:
        # store directly some z values
        # and the params / gradients norms
        logs['z'].append(z.detach()[::10])   <= This grows indefinitely on GPU RAM.
        logs['z_l2'].append(l2(z).item())

        logs['dictionary_sparsity'].append(l0_eps(model.get_dictionary()).mean().item())
        logs['dictionary_norms'].append(l2(model.get_dictionary(), -1).mean().item())

        for name, param in model.named_parameters():
            if param.grad is not None:
                logs[f'params_norm_{name}'].append(l2(param).item())
                logs[f'params_grad_norm_{name}'].append(l2(param.grad).item())
```

If you like I can make a PR to fix this @fel-thomas

**Comment by fel-thomas** [comment]
Ah, great catch, really sorry ^^
If you’re up for opening a quick PR with the fix (z.detach().cpu()[::10]), I’ll happily merge it. 
Thanks for taking the time to spot this! 😊

In fact, I’m even debating whether we should just remove that line entirely ^^ it’s kind of arbitrary and mostly useful for debugging, but maybe a bit out of scope for what the library should be doing by default, what do you think ?
